src/Modules/Data.lua

```diff
@@ -835,10 +835,13 @@ end
 data.gems = LoadModule("Data/Gems")
 data.gemForSkill = { }
 data.gemForBaseName = { }
+data.gemsByGameId = { }
 local function setupGem(gem, gemId)
 	gem.id = gemId
 	gem.grantedEffect = data.skills[gem.grantedEffectId]
 	data.gemForSkill[gem.grantedEffect] = gemId
+	data.gemsByGameId[gem.gameId] = data.gemsByGameId[gem.gameId] or {}
+	data.gemsByGameId[gem.gameId][gem.variantId] = gem
```

Comment:
Could we just make this map's inner key be `grantedEffectId`?  That way we don't need to add `variantId` to the save file format.  Then in `SkillsTab.lua` you can skip the if statement and loop and do `possibleVariants[child.attrib.skillId]`

Reply:
The reason I chose to use the variant id over the granted effect was that it better mapped to what we actually wanted to identify, the variant of the game. While right now it the primary granted effect is unique and does work as well, it may not always be the case. For example, they can be (accidentally) swapped so that what is now `GrantedEffect` could become `GrantedEffect2` and vice versa or if GGG decides to add "Awakened Arc" that grants both `Arc` and `ArcAltX` or similar.

All in all it just felt like the safer and most future-proof way to identify which gem was being used and since it is very desirable to support old save formats, picking the one least likely to change seemed like the path forward.

There is also the point of third party tools which parse and use our save format. If we could get some feedback from them it could help nail down the best format to use.

Reply:
Well reasoned, and if we're going to change our format, adding the variant explicitly seems like a solid way forward.  I'll throw the question in discord to the other third party tool devs about format